src/Model/EditorGroup.re

```diff
@@ -61,16 +61,23 @@ let getOrCreateEditorForBuffer = (state, bufferId) => {
   };
 };
 
-// TODO: Just use List.find_opt?
 let rec _getIndexOfElement = elem =>
   fun
-  | [] => (-1)
-  | [hd, ...tl] => hd === elem ? 0 : _getIndexOfElement(elem, tl) + 1;
+  | [] => None
+  | [hd, ...tl] =>
+    hd === elem
+      ? Some(0)
+      : (
+        switch (_getIndexOfElement(elem, tl)) {
+        | None => None
+        | Some(i) => Some(i + 1)
+        }
```

Comment:
You can use `Option.map` for this:

```suggestion
        _getIndexOfElement(elem, tl) |> Option.map(succ)
```

Reply:
I figured there had to be something for that like in Rust but when looking at the Reason docs I did not see much. I need to check the OCaml docs more I suppose. Good tip with using `succ` too.